platform-sdk/docs/proposals/TSS-Library/TSS-Library.md

```diff
@@ -0,0 +1,852 @@
+
+#### Summary diagram
+The following diagram exposes the steps necessary for bootstrapping and using the library. In orange are all the steps where the library is involved.
+The rest of the steps need to happen but are outside the scope of the described library.
+
+![img_2.png](img_2.svg)
+
+
+### Architecture
+To implement the functionality detailed in the previous section, the following code structure is proposed:
+
+![img_5.png](img_5.svg)
+
+1. **TSS Lib**: The consensus node will use the TSS library to create shares, create TSS messages to send to other nodes,
+   assemble shared public keys (ledgerId), and sign block hashes.
+2. **Bilinear Pairings Signature Library**: This library provides cryptographic objects (PrivateKey, PublicKey, and Signature)
+   and operations for the block node and consensus node to sign and verify signatures. The consensus node uses this library indirectly through the TSS Library.
+3. **Bilinear Pairings API**: An API definition for the cryptography primitives
+   and arithmetic operations required to work with a specific EC curve and the
+   underlying Groups, Fields, and Pairings. This API minimizes the impact of
+   changing to different curve implementations.
+4. **Bilinear Pairings**: An implementation of the Bilinear Pairings API that will be loaded at runtime using Java’s service provider (SPI) mechanism.
+   Multiple implementations can be provided to support different types of curves and
+   the Java service provider approach facilitates easily changing between
+   implementations and dependencies.
+6. **Native Support Lib**: Provides a set of generic functions loading native libraries in different system architectures when packaged in a jar
+   using a predefined organization so they can be accessed with JNI.
+6. **[Arkworks](https://github.com/arkworks-rs)**: A Rust ecosystem for cryptography. Our implementation uses it as the library responsible for elliptic curve cryptography.
+7. **EC-Key Utils** is a utility module that enables the node operator to generate a bootstrap public/private key pair.
+
+### Module organization and repositories
+![img_6.png](img_6.svg)
+1. **hedera-cryptography**: This is a separate repository for hosting cryptography-related libraries.
+   It is necessary to facilitate our build process, which includes Rust libraries. It also provides independent release cycles between consensus node code and other library users.
+2. **swirlds-native-support**: Gradle module that enables loading into memory compiled native libraries so they can be used with JNI.
+3. **swirlds-crypto-tss**: Gradle module for the TSS Library.
+   This library's only client is the consensus node, so it will be in the hedera-services repository, under the `platform-sdk` folder.
+4. **swirlds-crypto-signatures**: Gradle module for the Bilinear Pairings Signature Library.
```

Comment:
Is this the module that is a dependency for anyone using the ledger id to verify aggregated signatures? I.e. the block node when verifying that the network has agreed on and signed the block hash?

Reply:
Indeed.